app/controllers/devise/passwords_controller.rb

```diff
@@ -80,4 +85,11 @@ def unlockable?(resource)
     def translation_scope
       'devise.passwords'
     end
+
+  private
+    def expired_token_error(resource)
+      (resource.errors.respond_to?(:details) &&
```

Comment:
Could we break this check into smaller methods to improve readability? Also, would be good have some documentation explaining how we check whether the token is expired or not.

Reply:
I remembered that Devise 5 will drop Rails 4 support, so we don't need the conditional anymore. I guess it's simpler now and self-explanatory. WDYT?